### PR TITLE
[mdns]: Add unit test for services

### PR DESCRIPTION
**Pull request by david-cermak**
## Adds few test cases into mdns unit tests

* adding/removing services
* updating service properties (instance, port, txt)
* for selfhosted and delegated services

### CODE REVIEW

File: components/mdns/tests/unit_test/main/test_mdns.c [hunk 1]
```diff
@@ -142,12 +143,154 @@ TEST(mdns, query_api_fails_with_expected_err)
...
+    const size_t txt_data_cout = sizeof(txt_data) / sizeof(txt_data[0]);
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_service_txt_set(MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, txt_data, txt_data_cout));
+    yield_to_all_priorities();  // Make sure that mdns task has executed to add the hostname
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_lookup_selfhosted_service(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, 1, &results));
+    TEST_ASSERT_NOT_EQUAL(NULL, results);
+    TEST_ASSERT_NOT_EQUAL(NULL, results->txt);
+    // compare txt values by keys
+    size_t misses = 0;
+    for (int i = 0; i < txt_data_cout; ++i)
+        for (int j = 0; j < txt_data_cout; ++j)
```
[the remark this answers is not in the document]
Comment (david-cermak) [reply components/mdns/tests/unit_test/main/test_mdns.c:193]: I meant to only compare **values** with matching **keys**, need the two loops (tried to explain in one line above)
```cpp
    // compare txt values by keys
```

but you're right, one loop should iterate from 0 to `txt_data_cout`, while the other one the number of received results.

There's one more issue, later on, we shouldn't count misses, but matches to check correctly. will fix.

[the remark this answers is not in the document]
Comment (david-cermak) [reply components/mdns/tests/unit_test/main/test_mdns.c]: fixed and added some comments

File: components/mdns/tests/unit_test/main/test_mdns.c [hunk 2]
```diff
@@ -142,12 +143,154 @@ TEST(mdns, query_api_fails_with_expected_err)
...
+    };
+    const size_t txt_data_cout = sizeof(txt_data) / sizeof(txt_data[0]);
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_service_txt_set_for_host(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, MDNS_DELEGATE_HOSTNAME, txt_data, txt_data_cout));
+    yield_to_all_priorities();  // Make sure that mdns task has executed to add the hostname
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_lookup_delegated_service(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, 1, &results));
+    TEST_ASSERT_NOT_EQUAL(NULL, results);
+    TEST_ASSERT_NOT_EQUAL(NULL, results->txt);
+    // compare txt values by keys
+    size_t misses = 0;
+    for (int i = 0; i < txt_data_cout; ++i)
```
Comment (euripedesrocha) [inline components/mdns/tests/unit_test/main/test_mdns.c:265]: Shouldn't we iterate on results?

**Review by euripedesrocha — APPROVED** [review]

File: components/mdns/tests/unit_test/main/test_mdns.c (hunk 2 above)
Comment (david-cermak) [reply components/mdns/tests/unit_test/main/test_mdns.c:265]: could be that you've reviewed the old version of the file?

Comment (david-cermak) [reply components/mdns/tests/unit_test/main/test_mdns.c:265]: oh, I see. the same code for the delegated services. will fix

Comment (david-cermak) [reply components/mdns/tests/unit_test/main/test_mdns.c]: Updated the same way, thanks for noticing!
And thanks for the review, I'll merge it to proceed with the actual API fix.

**Pull request merged**